TuneBot/Modules/TrackModule.cs

```diff
@@ -2,331 +2,282 @@
+        public TrackModule(YouTubeService youtubeService, IConfiguration configuration, DbService dbService)
         {
-            this.youtube = youtube;
-            this.config = config;
+            _youtubeService = youtubeService;
+            _configuration = configuration;
+            _dbService = dbService;
         }
 
-        [Command("track")]
-        [Summary("Gets a track from Spotify and YouTube.")]
+        [SlashCommand("track", "Gets a track from Spotify and YouTube.")]
         public async Task GetTrack([Remainder] string details)
         {
             using var typing = Context.Channel.EnterTypingState();
 
-            var spotifyResultTask = SearchSpotify(details);
-            var youTubeResultTask = SearchYouTube(details);
-            await Task.WhenAll(spotifyResultTask, youTubeResultTask);
-           
-            await ReplyAsync(spotifyResultTask.Result ?? "No Spotify match found.");
-            await ReplyAsync("\n" + youTubeResultTask.Result ?? "No YouTube match found");
+            var spotifyResultTask = SearchSpotifyAsync(details);
+            var youTubeResultTask = SearchYouTubeAndGetLinkAsync(details);
+
+            var (spotifyResult, youtubeResult) = (await spotifyResultTask, await youTubeResultTask);
+
+            await RespondAsync(spotifyResult is null ? "No Spotify result found" : spotifyResult.Uri);
+            await RespondAsync(youtubeResult ?? "No YouTube result found");
         }
 
-        [Command("link")]
-        [Summary("Converts either a YouTube or Spotify URL.")]
+        [SlashCommand("link", "Converts either a YouTube or Spotify URL")]
         public async Task GetLink([Remainder] string link)
         {
             using var typing = Context.Channel.EnterTypingState();
             var result = "No match found.";
 
             if (link.Contains("spotify", StringComparison.OrdinalIgnoreCase))
             {
-                var spotifyTrack = await GetSpotifyTrack(link);
+                var spotifyTrack = await GetSpotifyTrackNameFromLink(link);
 
                 if (spotifyTrack != null)
-                    result = await SearchYouTube(spotifyTrack);
+                    result = await SearchYouTubeAndGetLinkAsync(spotifyTrack);
             }
             else if (link.Contains("youtube", StringComparison.OrdinalIgnoreCase) ||
-                    link.Contains("youtu.be", StringComparison.OrdinalIgnoreCase))
+                     link.Contains("youtu.be", StringComparison.OrdinalIgnoreCase))
             {
                 var youTubeResult = await GetYoutubeTitle(link);
 
                 if (youTubeResult != null)
                 {
-                    var spotifyResult = await SearchSpotify(youTubeResult);
+                    var spotifyResult = await SearchSpotifyAsync(youTubeResult);
 
                     if (spotifyResult != null)
-                        result = spotifyResult;
+                        result = spotifyResult.Uri;
```

Comment:
Same as above. This responds with spotify's internal URI of a track e.g. "spotify:track:0P1jQfRErBjgOSTbVKtCW7". Looking at the old code it to get the public URL it's track.ExternalUrls["spotify"]